internal/pkg/cli/interfaces.go

```diff
@@ -583,7 +583,7 @@ type runningTaskSelector interface {
 
 type dockerEngine interface {
 	CheckDockerEngineRunning() error
-	RedirectPlatform(string) (*string, error)
+	RedirectPlatform(string) (string, *string, error)
```

Comment:
🧐 The method signature is a bit unusual, it returns both a `string` and a `*string`. 

What do you think of the following series of refactors:
1. The package `dockerengine` should provide a `Platform()` method that only returns `(string, error)`. The platform that the docker daemon is using.
   The `dockerengine` object shouldn't know about what manifest platforms are supported.

2. The `manifest` pkg should own the logic to redirect platform if it can:
   ```go
   // RedirectPlatform returns a platform that's supported for the given manifest type.
   // If the container image os and arch is already supported, it returns the input string without any modifications.
   // Otherwise, if a platform cannot be redirected then an error is returned.
   func RedirectPlatform(imgPlatform, mftType string) (string, error)
   ```

Reply:
_A complete sidenote comment 📝_ 

We should write our method inputs and outputs irrespective of how a client might use it.
So for example, in the existing method we could do:
```go
RedirectPlatform(string) (imgPlatform, adjustedPlatform string, err error)
```
I understand that `*string` was returned because `manifest.PlatformString` is supposed to be a string pointer, but if there is more than one client for `RedirectPlatform` then the return type doesn't make sense.

Also returning a type that make sense for another pkg like `manifest.PlatformString` is a hint that this logic should live in the `manifest` pkg instead 😄